cf_units/__init__.py

```diff
@@ -508,14 +508,11 @@ def num2date(time_value, unit, calendar):
     unit = 'days since 001-01-01 00:00:00'}
     calendar = 'proleptic_gregorian'.
```

Comment:
The beginning of this paragraph (which I can't highlight!) is inaccurate since we're actually wrapping/mimicking `cftime.num2date`. If you can get this lined up we'll close #159 in the process 👍

Reply:
Thanks @trexfeathers, looks like the `maplotlib` behaviour has changed since #159 was opened.  How's this?

Reply:
mpl ref for info: https://matplotlib.org/stable/api/dates_api.html#matplotlib-date-format

Reply:
Except it's still not _quite_ the same since `cftime.num2date` returns a subclass of `datetime.datetime`.  Also we only work at second precision whereas `matplotlib` works at microsecond precision.

We could just remove this paragraph entirely?  Any user who is concerned about how our behaviour compares to `matplotlib` behaviour can always do their own comparison.

Reply:
https://github.com/SciTools/cf-units/pull/168#discussion_r647389483